src/tasks/leveling.ts

```diff
@@ -754,33 +809,36 @@ export const LevelingQuest: Quest = {
       limit: { tries: 12 },
     },
     {
-      name: "Use Ointment of the Occult",
+      name: "Use Reagent Booster",
       completed: () =>
-        (!have($item`grapefruit`) && !have($item`ointment of the occult`)) ||
-        have($effect`Mystically Oiled`),
+        (!have(reagentBoosterIngredient) && !have(reagentBoosterItem)) ||
+        have(reagentBoosterEffect),
       do: (): void => {
-        if (!have($item`ointment of the occult`)) {
-          if (get("reagentSummons") === 0) useSkill($skill`Advanced Saucecrafting`, 1);
-          create($item`ointment of the occult`, 1);
+        print(
+          `Completed condition: ${!have(reagentBoosterIngredient) && !have(reagentBoosterItem)}`
+        );
+        if (!have(reagentBoosterItem)) {
+          create(reagentBoosterItem, 1);
         }
-        ensureEffect($effect`Mystically Oiled`);
+        ensureEffect(reagentBoosterEffect);
       },
     },
     {
-      name: "Use Oil of Expertise",
+      name: "Use Reagent Balancer",
       ready: () => get("_loveTunnelUsed") || !get("loveTunnelAvailable"),
       completed: () =>
-        (!have($item`cherry`) && itemAmount($item`oil of expertise`) <= 1) ||
-        have($effect`Expert Oiliness`),
+        (!have(reagentBalancerIngredient) && itemAmount(reagentBalancerItem) <= 1) ||
+        have(reagentBalancerEffect) ||
+        itemAmount(reagentBalancerItem) === 1,
       do: (): void => {
-        if (!have($item`oil of expertise`)) {
+        if (!have(reagentBalancerItem)) {
           if (get("reagentSummons") === 0) useSkill($skill`Advanced Saucecrafting`, 1);
-          create($item`oil of expertise`, 1);
+            create(reagentBalancerItem, 1);
         }
-        if (itemAmount($item`oil of expertise`) > 1)
-          use($item`oil of expertise`, itemAmount($item`oil of expertise`) - 1);
-        if (have($item`cherry`) && have($effect`Expert Oiliness`))
-          putCloset(itemAmount($item`cherry`), $item`cherry`);
+        if (itemAmount(reagentBalancerItem) > 1)
+          use(reagentBalancerItem, itemAmount(reagentBalancerItem) - 1);
+        if (have(reagentBalancerIngredient) && have(reagentBalancerEffect))
+          putCloset(itemAmount(reagentBalancerIngredient), reagentBalancerIngredient);
       },
       limit: { tries: 1 },
     },
```

Comment:
How are mox classes going to get the olive for their reagent balancer?

Reply:
I... don't know. :) This probably won't work for Moxie until I figure that out.

Reply:
Is it possible to add buying a dynamic range at this point of the script? Also, advanced saucecrafting is not being cast for some reason